### PR TITLE
Feature/update task

### PR DESCRIPTION
**Pull request by nik0laevvlad**

### CODE REVIEW

File: src/modules/ListItem.tsx [hunk 1]
```diff
@@ -7,10 +7,11 @@ interface Props {
   item: TodoItem;
   completeTask: (selectedTodo: TodoItem) => void;
   deleteTask: (index: number) => void;
+  show: () => void;
 }
 
 export const ListItem = (
-  { item, completeTask, deleteTask }: Props,
+  { item, completeTask, deleteTask, show }: Props,
   key: number,
```
Comment (KyMback) [inline src/modules/ListItem.tsx]: why we need this second arg ?

File: src/modules/TodoList.tsx [hunk 2]
```diff
@@ -1,28 +1,79 @@
...
+                completeTask={completeTask}
+                deleteTask={() => deleteTask(key)}
+                show={() => show(key)}
+              />
+            </div>
           );
         })}
       </div>
-    </div>
+      <Form
```
Comment (KyMback) [inline src/modules/TodoList.tsx]: let's create separate component for item updating

File: src/modules/TodoList.tsx [hunk 3]
```diff
@@ -1,28 +1,79 @@
...
 }
 
-export const TodoList = ({ completeTask, list, deleteTask }: Props) => {
+export const TodoList = ({
+  completeTask,
+  list,
+  deleteTask,
+  updateTask,
+}: Props) => {
+  const [newText, setNewText] = useState('');
```
Comment (KyMback) [inline src/modules/TodoList.tsx]: you can incapsulate newText management in them

File: src/modules/TodoList.tsx [hunk 4]
```diff
@@ -1,28 +1,79 @@
...
+                show={() => show(key)}
+              />
+            </div>
           );
         })}
       </div>
-    </div>
+      <Form
+        hidden={hidden}
+        onSubmit={(e) => {
```
Comment (KyMback) [inline src/modules/TodoList.tsx]: and push data on the ancestor component only after submit

File: src/modules/TodoList.tsx [hunk 5]
```diff
@@ -1,28 +1,79 @@
...
+          <Col md="auto">
+            <FormControl
+              placeholder="update"
+              value={newText}
+              onChange={(e) => setNewText(e.target.value)}
+            />
+          </Col>
+          <Col md="auto">
+            <Button
+              onClick={() => {
```
Comment (KyMback) [inline src/modules/TodoList.tsx]: why you need here handler and in Form ?

File: src/modules/TodoList.tsx [hunk 6]
```diff
@@ -1,28 +1,79 @@
...
+    } else {
+      setHidden(false);
+    }
+  };
+
   return (
-    <div>
+    <>
       <div className="mt-3 h2">
         {list.map((item: TodoItem, key: number) => {
```
Comment (KyMback) [inline src/modules/TodoList.tsx]: it's not key it's index

File: src/App.tsx [hunk 7]
```diff
@@ -23,6 +23,13 @@ function App() {
     setList([...list]);
   };
 
+  const updateTask = (selectedTodo: TodoItem, text: string) => {
+    if (text.trim() !== '') {
```
Comment (KyMback) [inline src/App.tsx:27]: what if we want to remove text from the item ?

**Review by KyMback — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 7 above)
Comment (nik0laevvlad) [reply src/App.tsx:27]: why would we need an empty todo item?

File: src/modules/ListItem.tsx (hunk 1 above)
Comment (nik0laevvlad) [reply src/modules/ListItem.tsx]: removed

File: src/modules/TodoList.tsx (hunk 6 above)
Comment (nik0laevvlad) [reply src/modules/TodoList.tsx]: updated

File: src/modules/TodoList.tsx (hunk 5 above)
Comment (nik0laevvlad) [reply src/modules/TodoList.tsx]: updated

File: src/modules/TodoList.tsx [hunk 8]
```diff
@@ -1,28 +1,54 @@
...
+  list,
+  deleteTask,
+  updateTask,
+}: Props) => {
+  const [id, setId] = useState<number>();
+
+  const show = (key: number) => {
+    if (key !== id) {
+      setId(key);
+    }
```
Comment (KyMback) [inline src/modules/TodoList.tsx]:
```ts
if (key !== id) {
      setId(key);
} else {
    setId(undefined);
}
```

File: src/modules/TodoList.tsx [hunk 9]
```diff
@@ -1,28 +1,54 @@
...
 }
 
-export const TodoList = ({ completeTask, list, deleteTask }: Props) => {
+export const TodoList = ({
+  completeTask,
+  list,
+  deleteTask,
+  updateTask,
+}: Props) => {
+  const [id, setId] = useState<number>();
```
Comment (KyMback) [inline src/modules/TodoList.tsx]: mb selectedId, setSelectedId ?

**Review by KyMback — APPROVED** [review]

File: src/modules/TodoList.tsx (hunk 8 above)
Comment (nik0laevvlad) [reply src/modules/TodoList.tsx]: updated

File: src/modules/TodoList.tsx (hunk 9 above)
Comment (nik0laevvlad) [reply src/modules/TodoList.tsx]: updated

**Pull request merged**